service/src/main/java/uk/gov/hmcts/reform/fpl/service/RespondentMigrationService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class RespondentMigrationService {
+
+    public AboutToStartOrSubmitCallbackResponse setMigratedValue(CaseDetails caseDetails) {
+        Map<String, Object> data = caseDetails.getData();
+
+        if (caseDetails.getData().containsKey("respondents1") || !caseDetails.getData().containsKey("respondents")) {
+            data.put("respondentsMigrated", "Yes");
+
+            if (!caseDetails.getData().containsKey("respondents1")) {
+                List<Map<String, Object>> populatedRespondent = new ArrayList<>();
+                populatedRespondent.add(ImmutableMap.of(
+                    "id", UUID.randomUUID().toString(),
+                    "value", ImmutableMap.of(
+                        "party", ImmutableMap.of(
+                            "partyId", UUID.randomUUID().toString()
+                        )
+                    ))
+                );
+                data.put("respondents1", populatedRespondent);
+            }
+
+            return AboutToStartOrSubmitCallbackResponse.builder()
+                .data(data)
+                .build();
+        } else {
+            data.put("respondentsMigrated", "No");
```

Comment:
Wouldn't be better to default to `Yes` so we capture details in new format as soon as possible and migrate old cases later?

If `No` is a default option then any case created after migration will again have old structure and another migration will be needed.

Am I missing something here?

Reply:
Ignore me, I misread it. Your test helped me understand behaviour.